### PR TITLE
WIP: Add option to install ReShade shaders

### LINKED ISSUE #2: Add option to install ReShade shaders
Hey again, just wanted to let you know that it _seems_ like at least some preset/shader authors in the community (and probably all of them as time goes forward) are gearing their new releases around the assumption that the user just straight installed ReShade 5.6 while checking all the "install shaders" options. I.e. there's an assumption that `reshade-shaders` has everything in it that a normal ReShade 5.6 install would have (and only those to an extent).

As I've found, however, what GShade had is not necessarily the same as the above and so you may run into weird and exciting breakages from those differences.

So with that in mind, future-proofing wise it may be desirable to be able to only install the base `reshade-shaders` as done by the ReShade installation process. From here: https://github.com/kevinlekiller/reshade-steam-proton/blob/main/reshade-linux.sh#L305
It seems like a full installation wherein you checked all the optional shaders has the following repos merged into one `reshade-shaders/{Shaders,Textures}`, with some of the optional extras as sub-folders for the `Shaders/` case.
```
https://github.com/CeeJayDK/SweetFX
https://github.com/martymcmodding/qUINT
https://github.com/BlueSkyDefender/AstrayFX
https://github.com/prod80/prod80-ReShade-Repository
https://github.com/crosire/reshade-shaders
```
I got a friend on Windows to do a fresh install to confirm the directory structure and at least by default it seems that `reshade-shaders/Shaders` does the following:
- All files under `Shaders` in their respective git repos are copied directly to `reshade-shaders/Shaders` for the following:
  - SweetFX
  - reshade-shaders
- All files under `Shaders` in their respective git repos are copied _into a subdirectory_ within `reshade-shaders/Shaders` for the following:
  - qUINT - Subdirectory: `reshade-shaders/Shaders/qUINT/`
  - AstrayFX - Subdirectory: `reshade-shaders/Shaders/AstrayFX` (including another subdirect `OneShot` as in the repo)
  - prod80-ReShade-Repository - Subdirectory: `reshade-shaders/Shaders/PD80/`

The `Textures` directory remains flat, just a direct copy of all `Textures/` folders from the above five repos.

If you wanted to be really fancy, as they're all git repositories, you could optionally depend upon `git` and `git clone <url>` all five of those repos (perhaps as shallow copies on first install), symlink them into the merged structure (as the kevinlekiller script does), and then use that structure accordingly. That way update operations can be as simple as `git pull` on all of them (or be even more fancy and setup the download area as a git repo that those 5 are subrepos of and then use a submodule recursive update to do all of them, that seems like overkill though).

Alternatively just straight zip download as per usual of course.

As a plus to all this, all involved repositories don't have asinine unenforceable provisions against downloading them, and some of them are even open source, so this can be automated.

Oh also as an additional bonus, here's the full text of the default `ReShade.ini` configuration file created for this type of installation (the subdirectories in `reshade-shaders/Shaders` are included explicitly rather than by using a wildcard catch all)
```
[APP]
ForceDefaultRefreshRate=0
ForceFullscreen=0
ForceVsync=0
ForceWindowed=0

[GENERAL]
EffectSearchPaths=.\reshade-shaders\Shaders,.\reshade-shaders\Shaders\qUINT,.\reshade-shaders\Shaders\PD80,.\reshade-shaders\Shaders\AstrayFX
PreprocessorDefinitions=RESHADE_DEPTH_LINEARIZATION_FAR_PLANE=1000.0,RESHADE_DEPTH_INPUT_IS_UPSIDE_DOWN=0,RESHADE_DEPTH_INPUT_IS_REVERSED=1,RESHADE_DEPTH_INPUT_IS_LOGARITHMIC=0
TextureSearchPaths=.\reshade-shaders\Textures

[INPUT]
GamepadNavigation=1
KeyOverlay=36,0,0,0
```

### PR DESCRIPTION
**Pull request by cozyGalvinism**
Installing ReShade also installs a ReShade.ini now

### CODE REVIEW

File: src/lib/lib.rs [hunk 1]
```diff
@@ -43,6 +107,76 @@ pub async fn download_file(client: &reqwest::Client, url: &str, path: &str) -> R
...
+        }
+
+        if repo_shader_directory.exists() {
+            let builder = CopyBuilder::new(&repo_shader_directory, &target_directory);
+            builder.overwrite(true).run()?;
+        }
+
+        if repo_texture_directory.exists() {
+            let builder = CopyBuilder::new(&repo_texture_directory, &texture_directory);
+            builder.overwrite(true).run()?;
```
Comment (ultimatespirit) [inline src/lib/lib.rs:152]: We run into an ordering issue here between `crosire/reshade-shaders` and `CeeJayDK/SweetFX`, where if one repo had overlapping files that were not exactly the same then the order in which we add them matters. It stands to reason that it's likely for `crosire/reshade-shaders` to be the base directory over which things get overwritten, if needed. Currently I believe the ordering of repos in `SHADER_REPOSITORIES` makes it so `SweetFX` would get overwritten.

And indeed, if you run the following (assuming all pulled repositories are in the CWD at start)....
```
$ cd SweetFX
$ shasum Shaders/* > ../SweetFX-Shaders-SHAsums.txt
$ cd ../reshade-shaders
$ shasum -qc ../SweetFX-Shaders-SHAsums.txt --ignore-missing
Shaders/FXAA.fx: FAILED
Shaders/Nostalgia.fx: FAILED
Shaders/SMAA.fx: FAILED
shasum: WARNING: 3 computed checksums did NOT match
```
So at time of writing three files in `SweetFX` that are also in `crosire/reshade-shaders` on the `master` branch do *not* match.

I checked the SHAs versus a copy of `reshade-shaders` from a fresh install on Windows that a friend provided me and `SweetFX` shaders match perfectly, whereas `crosire/reshade-shaders` did not match for 6 files (the 3 noted above that conflict between SweetFX and itself, as well as 3 additional files that are newer in the fresh install than what appears to be currently on `master` in `crosire/reshade-shaders`. These files appear to be from the `slim` branch.).

For what it's worth, there is _no_ overlap currently between `SweetFX` and the `slim` branch, and it seems none of the files on `master` that are not in `slim` are in the fresh install. So I think this may be a case of the infrastructure having changed since the time of the shell script my original suggestion was based off of. Indeed that script actually didn't even make the subdirectories currently seen, it just shoved everything into one flat `Shaders` directory, though it did not overwrite anything.

Tl;dr it seems we need to use the `slim` branch to conform most closely to what a pristine installation does, and regardless we want `crosire/reshade-shaders` to go first / get overwritten when a conflict emerges. If we did want to ship the extra shaders `master` has over `slim` it would require some finangling since `slim` is definitely newer than `master` is currently.

Also having said all that... it seems this file is probably our canonical information sheet on what to install and how: https://github.com/crosire/reshade-shaders/blob/list/EffectPackages.ini

From the `list` branch of `crosire/reshade-shaders`.

So... that's an oops on my part sorry.

File: reshade.example.ini [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[APP]
+ForceDefaultRefreshRate=0
+ForceFullscreen=0
+ForceVsync=0
+ForceWindowed=0
+
+[GENERAL]
+EffectSearchPaths=.\reshade-shaders\Shaders,.\reshade-shaders\Shaders\qUINT,.\reshade-shaders\Shaders\PD80,.\reshade-shaders\Shaders\AstrayFX,.\gshade-shaders\Shaders,.\gshade-shaders\ComputeShaders
```
Comment (ultimatespirit) [inline reshade.example.ini:8]: We may not want to by default allow both `reshade-shaders` and `gshade-shaders`. If someone ends up with both directories in their area for whatever reason shaders will double load which is bad. Same story for textures.

**Review by ultimatespirit — COMMENTED** [review]
I think this will end up as a review on this merged pull rather than an Issue, but really what I've brought up are Issues at this point... I'll copy them over to an issue if needed.
